src/pages/index.js

```diff
@@ -0,0 +1,331 @@
+const profileSubmitButton = editProfileModal.querySelector(
+  ".modal__submit-button"
+);
+
+const cardNameInput = addCardModal.querySelector("#add-card-name-input");
+const cardLinkInput = addCardModal.querySelector("#add-card-link-input");
+
+const previewModal = document.querySelector("#preview-modal");
+const previewModalImageEl = previewModal.querySelector(".modal__image");
+const previewModalCaptionEl = previewModal.querySelector(".modal__caption");
+const previewModalCloseButton = previewModal.querySelector(
+  ".modal__close-button"
+);
+
+const editModalCloseButton = editProfileModal.querySelector(
+  ".modal__close-button"
+);
+
+const editModalNameInput = editProfileModal.querySelector(
+  "#profile-name-input"
+);
+const editModalDescriptionInput = editProfileModal.querySelector(
+  "#profile-description-input"
+);
+
+const cardTemplate = document.querySelector("#card-template");
+
+const cardsList = document.querySelector(".cards__list");
+
+const avatarButton = document.querySelector(".profile__avatar-button");
+const addAvatarModal = document.querySelector("#avatar-modal");
+const avatarForm = document.querySelector("#edit-avatar-form");
+const avatarInput = document.querySelector("#avatar-input");
+const avatarModalCloseButton = addAvatarModal.querySelector(
+  ".modal__close-button"
+);
+const avatarSubmitButton = addAvatarModal.querySelector(
+  ".modal__submit-button"
+);
+
+const deleteModal = document.querySelector("#delete-modal");
+const deleteForm = deleteModal.querySelector(".modal__form");
+const deleteCancelButton = deleteModal.querySelector("#cancel-delete-button");
+const deleteCloseButton = deleteModal.querySelector(".modal__close-button");
+
+let selectedCard;
+let selectedCardId;
+
+function getCardElement(data) {
+  const cardElement = cardTemplate.content
+    .querySelector(".card")
+    .cloneNode(true);
+
+  const cardNameEl = cardElement.querySelector(".card__title");
+  const cardImageEl = cardElement.querySelector(".card__image");
+  const cardLikeButton = cardElement.querySelector(".card__like-button");
+  const cardDeleteButton = cardElement.querySelector(".card__delete-button");
+
+  cardNameEl.textContent = data.name;
+  cardImageEl.src = data.link;
+  cardImageEl.alt = data.name;
+
+  if (data.isLiked) {
+    cardLikeButton.classList.add("card__like-button_liked");
+  }
+
+  cardLikeButton.addEventListener("click", (evt) => {
+    handleCardLike(evt, data._id);
+  });
+
+  cardDeleteButton.addEventListener("click", () => {
+    handleDeleteCard(cardElement, data._id);
+  });
+
+  cardImageEl.addEventListener("click", () => {
+    handleImageClick(data);
+  });
+
+  return cardElement;
+}
+
+function handleCardLike(evt, dataId) {
+  let isLiked = evt.target.classList.contains("card__like-button_liked")
+    ? true
+    : false;
+  api
+    .toggleLike(dataId, isLiked)
+    .then(evt.target.classList.toggle("card__like-button_liked"))
```

Comment:
`    .then(evt.target.classList.toggle("card__like-button_liked"))
`

In `then`, `catch`, `finally` you need to pass only the **reference** to the function (not a function call) otherwise the function is called immediately without waiting for the response from the server. You can pass an **unnamed** arrow function as well.

Reply:
I fixed this and removed other ()=> calls. Didn't seem to break anything, so hopefully it's what you wanted.

Reply:
**()=> calls** are needed. 

It should be either `()=> calls` or **named functions**